starfish/core/_display.py

```diff
@@ -190,49 +194,52 @@ def display(
     Notes
     -----
     - To use in ipython, use the `%gui qt5` magic.
-    - Napari axes currently cannot be labeled. Until such a time that they can, this function will
+    - napari axes currently cannot be labeled. Until such a time that they can, this function will
       order them by Round, Channel, and Z.
-    - Requires napari 0.0.6: install starfish using `pip install starfish[napari]` to install all
+    - Requires napari 0.0.9: install starfish using `pip install starfish[napari]` to install all
       necessary requirements
     """
     if stack is None and spots is None:
+        # masks without stack or spots have no context so don't check for that
         raise TypeError("expected a stack and/or spots; got nothing")
 
-    if Window is None or Viewer is None:
-        warnings.warn("Requires napari 0.0.6. Run `pip install starfish[napari]` to install the "
-                      "necessary requirements.")
-        return
-
-    from PyQt5.QtWidgets import QApplication
+    try:
+        import napari
+        if napari.__version__ != NAPARI_VERSION:
+            raise Exception
+    except Exception:
+        raise ImportError(f"Requires napari {NAPARI_VERSION}. "
+                           "Run `pip install starfish[napari]` "
+                           "to install the necessary requirements.")
 
-    # Instantiate the napari viewer
+    # instantiate the napari viewer
     if viewer is None:
+        from qtpy.QtWidgets import QApplication
         app = QApplication.instance() or QApplication([])
 
-        # initialize the viewer
         viewer = Viewer()
-        window = Window(viewer, show=False)
-        viewer._window = window
         new_viewer = True
     elif isinstance(viewer, Viewer):
         new_viewer = False
     else:
-        raise TypeError("viewer must be a napari Viewer or None")
+        raise TypeError("viewer must be a napari.Viewer or None")
 
     if stack is not None:
         if project_axes is not None:
             stack = stack.max_proj(*project_axes)
 
         # Switch axes to match napari expected order [x, y, round, channel, z]
         reordered_array: np.ndarray = stack.xarray.transpose(
-            Axes.Y.value,
-            Axes.X.value,
-            Axes.ROUND.value,
             Axes.CH.value,
-            Axes.ZPLANE.value
+            Axes.ROUND.value,
+            Axes.ZPLANE.value,
+            Axes.Y.value,
+            Axes.X.value
         ).values
 
-        layer = viewer.add_image(np.squeeze(reordered_array), multichannel=False)
+        layer = viewer.add_image(np.squeeze(reordered_array),
+                                 multichannel=False,
+                                 name='stack')
         layer.colormap = "gray"
```

Comment:
now colormap default is gray, but can also pass `colormap` keyword arg to the above add_image function